metrics/deploy/docs/diagrams/deploy_tool_class_diagram.puml

```diff
@@ -0,0 +1,85 @@
+@startuml deploy_tool_class_diagram
+
+package cli {
+
+    package flutter {
+        class FlutterCommand {
+            + version(): Future<void>
+        }
+    }
+
+    package firebase {
+        class FirebaseCommand {
+            + login(): Future<String>
+            + addFirebase(String projectID, String firebaseToken): Future<void>
+            + createWebApp(String projectID, String firebaseToken): Future<String>
+            + downloadSDKConfig(String appID, String configPath, String projectID, String firebaseToken): Future<void>
+            + version(): Future<void>
+        }
+    }
+
+    package gcloud {
+        class GCloudCommand {
+            - _chars : String
+            - _rnd : Random
+            + projectID : String
```

Comment:
It would be great to declare public members above the private ones.
```suggestion
            + projectID : String
            - _chars : String
            - _rnd : Random
```